R/request_handler-httr2.R

```diff
@@ -81,7 +81,29 @@ httr2_body <- function(x) {
     x$body$type,
     raw = {
       # httr2::req_body_raw allows raw or string
-      if (is_raw(x$body$data)) rawToChar(x$body$data) else x$body$data
+      if (has_binary_content(x$headers)) {
+        tryCatch(
```

Comment:
I think you could probably simplify the rest of this now

Reply:
I think you could make that even simpler. Like keep it as raw if it has binary content, otherwise convert it to text.